src/qtils/error_throw.hpp

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright Quadrivium LLC
+ * All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <system_error>
+
+#include <boost/throw_exception.hpp>
+
+namespace qtils {
+  using ErrorException = boost::wrapexcept<std::system_error>;
+
+  /**
+   * @brief throws outcome::result error as a boost exception
+   * @tparam T enum error type, only outcome::result enums are allowed
+   * @param t error value
+   */
+  template <typename T>
+    requires std::is_enum_v<T>
+  void raise(T t) {
+    std::error_code ec = make_error_code(t);
+    boost::throw_exception(std::system_error(ec));
+  }
```

Comment:
There is [qtils::asError](https://github.com/qdrvm/qtils/blob/3c2828ffbda0dfd1b4d3de907a3fbfb0d447526b/src/qtils/error.hpp#L28-L35) conversion function.